deeplinkdispatch/src/main/java/com/airbnb/deeplinkdispatch/DeepLinkEntry.java

```diff
@@ -27,76 +31,75 @@
   private static final String PARAM = "([a-zA-Z][a-zA-Z0-9_-]*)";
   private static final String PARAM_REGEX = "\\{(" + PARAM + ")\\}";
 
-  private final String hostPath;
+  final Type type;
+  final Class<?> activityClass;
+  final String method;
+  private final Set<String> parameters;
   private final String regex;
-  private final Type type;
-  private final Class<?> activityClass;
-  private final String method;
 
-  public enum Type {
+  enum Type {
     CLASS,
     METHOD
   }
 
-  public DeepLinkEntry(String hostPath, Type type, Class<?> activityClass, String method) {
-    this.hostPath = hostPath;
-    this.regex = generateLookupString(hostPath);
+  DeepLinkEntry(String uri, Type type, Class<?> activityClass, String method) {
     this.type = type;
     this.activityClass = activityClass;
     this.method = method;
+    this.parameters = parsePathParameters(uri);
+    this.regex = schemeHostAndPath(uri).replaceAll(PARAM_REGEX, PARAM_VALUE);
   }
 
-  public String getRegex() {
```

Comment:
Not sure why we got rid of the getter/setters here. I understand that it's still accessible as a member variable, but my preference is that it stays accessed through these methods.